bluebridge-app.js

```diff
@@ -75,20 +79,35 @@ Polymer({
         });
     },
 
+    _firebaseStatusKnownChanged: function _firebaseStatusKnownChanged(newStatus) {
+        if (newStatus) {
+            console.log('[bluebridge-app] Firebase is ready for action');
+
+            var currentUser = firebase.auth().currentUser;
+
+            if (currentUser) {
+                this._setInternalFirebaseToken(currentUser);
+            }
+        }
+    },
+
     _firebaseUidChanged: function _firebaseUidChanged(uid) {
         var promise = null;
 
+        // No point checking uid if firebase isn't even available
+        // for us to check
+        if (!this.firebaseStatusKnown) {
+            console.warn('[bluebridge-app] Firebase not ready when UID changed');
+            return;
+        }
+
         if (uid === undefined || !uid) {
-            promise = firebase.auth().signInAnonymously()
-            //.then(this._setInternalFirebaseToken.bind(this))
-            .catch(function (err) {
-                return Promise.reject('Failed to sign in anonymously ' + err);
-            });
-        } else {
-            promise = this._setInternalFirebaseToken(firebase.auth().currentUser);
+            // Anonymous sign-in will kick off a uid change, so we will
```

Comment:
I do not think we should sign-in anonymously here. We should check when firebase status known is true.